providers/provtarget/builddefaults/default.go

```diff
@@ -113,11 +113,9 @@ func (b *Resolver) Targets(ctx context.Context, log *logrus.Entry) []string {
 	return servers.Strings()
 }
 
-// setBuildBasedOnJWT sets build settings based on contents of a JWT file
-func (b *Resolver) setBuildBasedOnJWT() (*tokens.ProvisioningClaims, error) {
-	bi := b.bi
-
-	d, err := os.ReadFile(bi.ProvisionJWTFile())
+// SetBuildBasedOnJWT sets build settings based on contents of a JWT file
+func SetBuildBasedOnJWT(jwtf string, bi *build.Info, readFilef func(string) ([]byte, error)) (*tokens.ProvisioningClaims, error) {
```

Comment:
Hmm, think its a bit non idiomatic with the readFilef here.

Typically you'd do like `SetBuildBasedOnJWT(bi *build.Info, jwt string)` I get that it would probably help a bit in callers to make their code shorter and more generic but would rather do like this.

An alternative would be something like this

```go
func SetBuildBasedOnJWTFile(filesys fs.FS,path string){
	f,_:=filesys.Open(path)
	jwt,_:=io.ReadAll(f)

	// do stuff
}
```

You then do:

```
SetBuildBasedOnJWTFile(os.DirFS("/tmp"), "x.jwt")
```

Of if you have an embedded fs you do `SetBuildBasedOnJWTFile(internal.FS, "x.jwt")` or similar, I quite like this.

Reply:
Will need an issue opened to help me with changelogs - you can also update chnagelog at the same time if you want (i always forget so no biggie)

Reply:
Cool, I'll take a swing at that.